### PR TITLE
Content Loader Added & Webpack Migrated to V4

### PR DESCRIPTION
**Pull request by haideralishah**
Added content loader to all pages.

Webpack migrated to v4

Merged latest code from develop to feature/content_loader branch.

### CODE REVIEW

File: ios/TreecounterApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -58,6 +57,7 @@
 		ED297163215061F000B7C4FE /* JavaScriptCore.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = ED297162215061F000B7C4FE /* JavaScriptCore.framework */; };
 		FA70D9E8C5764BEE8D9A91FF /* OpenSans-RegularItalic.ttf in Resources */ = {isa = PBXBuildFile; fileRef = C297B0E8F9D447E9921BBBEC /* OpenSans-RegularItalic.ttf */; };
 		FE9B30FE88034D31A30B7F1E /* OpenSans-Regular.ttf in Resources */ = {isa = PBXBuildFile; fileRef = CED59F86B5B44DAAB29EF987 /* OpenSans-Regular.ttf */; };
+		C049270B2B22481BBB9AA7E1 /* libRNPaypalWrapper.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 7A8A6D0B59B54553B2985F37 /* libRNPaypalWrapper.a */; };
```
Comment (norbertschuler) [inline ios/TreecounterApp.xcodeproj/project.pbxproj]: Watch out, you added the paypal libs to iOS (which will lead to our iOS app being rejected from Apple in the review process). Please merge the latest develop branch where this npm module got removed. Unfortunately `react-native link` cannot removed this additions to the iOS project anymore afterwards, so you have to copy a fresh version from the develop branch and link again and remove this stuff manually. Compare with the latest discussion in our #app Slack channel.

File: package.json [hunk 2]
```diff
@@ -35,8 +35,9 @@
     "bugsnag-react-native": "^2.23.2",
     "classnames": "^2.2.5",
     "css-loader": "^0.28.9",
-    "date-fns": "^2.6.0",
-    "esri-loader": "^2.12.0",
+    "d3-interpolate": "^1.3.2",
+    "date-fns": "^2.4.1",
```
Comment (norbertschuler) [inline package.json]: why this downgrade of `date-fns`?

File: package.json [hunk 3]
```diff
@@ -68,7 +70,8 @@
     "react-native-lightbox": "^0.7.0",
     "react-native-linear-gradient": "^2.5.4",
     "react-native-material-dropdown": "git+https://github.com/niteshgrg/react-native-material-dropdown.git",
-    "react-native-permissions": "^1.2.1",
+    "react-native-paypal-wrapper": "^1.3.2",
+    "react-native-permissions": "^1.2.0",
```
Comment (norbertschuler) [inline package.json]: another downgrade?

File: web/config/webpack.dev.config.js [hunk 4]
```diff
@@ -37,12 +44,24 @@ module.exports = webpackMerge(commonConfig, {
   devtool: 'source-map',
   devServer: {
     inline: true,
-    port: 8080,
+    port: 8081,
     historyApiFallback: true,
     disableHostCheck: true,
-    host: '0.0.0.0'
+    host: '192.168.100.23'
```
Comment (norbertschuler) [inline web/config/webpack.dev.config.js]: please do not submit your local configuration

Comment (haideralishah) [reply web/config/webpack.dev.config.js]: Sure.

File: ios/TreecounterApp.xcodeproj/project.pbxproj (hunk 1 above)
Comment (haideralishah) [reply ios/TreecounterApp.xcodeproj/project.pbxproj]: Okey. I do so.

**Comment by norbertschuler** [comment]
@haideralishah Personally I would prefer separating the pull requests for implementing content-loader from upgrading to webpack version 4. Maybe you could create to branches and two pull requests to avoid one feature not being able to merge because of the other one having issues?

**Comment by haideralishah** [comment]
Alright.

**Comment by haideralishah** [comment]
I am making another pull request for webpack migration. Kindly discard this one.

**Comment by norbertschuler** [comment]
> I am making another pull request for webpack migration. Kindly discard this one.

I won't want to delete your work on content-loader. Do you also want to bring that back in another PR?

**Comment by norbertschuler** [comment]
Probably this got replace now by #1637 and #1625

**Pull request closed without merging**